parser.c

```diff
@@ -19,6 +19,7 @@ static Dict *struct_defs = &EMPTY_DICT;
 static Dict *union_defs = &EMPTY_DICT;
 static List *localvars = NULL;
 
+static Ctype *ctype_void = &(Ctype){CTYPE_VOID, 1, NULL};
```

Comment:
Please be aware of the fact that type` void` has no size. 

quoting N1570 6.5.3.4 paragraph 1:
> The sizeof operator shall not be applied to an expression that has function type or an incomplete type, to the parenthesized name of such a type, or to an expression that designates a bit-field member.

The assignment to `1` might violate the properties of `void` though.

Reply:
I had found this negligence before I committed. But I found[ GCC C Extension 5.20](https://gcc.gnu.org/onlinedocs/gcc-4.4.2/gcc/Pointer-Arith.html#Pointer-Arith) quoting that:

In GNU C, addition and subtraction operations are supported on pointers to void and on pointers to functions. This is done by treating the size of a void or of a function as 1.
A consequence of this is that sizeof is also allowed on void and on function types, and returns 1.`

Should I follow GCC or consider sizeof is disallowed on void and function type.

Reply:
We can stick on C standard and mention GNU C extension somewhere in source code.

Reply:
I found there doesn't exist sizeof in this project. I know the necessity of sizeof, but I don't know the order? (e.g. If I need to build sizeof first?)

Reply:
You don't have to implement `sizeof` support at first glance.

Reply:
I think we can use size = 1 on `void`, just handling void type when parsing sizeof. Since the approach can let `void *` have same behavior with `char *`.

Reply:
Add some informative comments here.

Reply:
Sorry, I misunderstood the `emit_pointer_arith`, function in `codegen_x64.c`. I am embarrassed about the mistake. Using zero size for `void` is better.